FeedStoreChallenge/ManagedCache.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  ManagedCache.swift
+//  FeedStoreChallenge
+//
+//  Created by Darren Findlay on 29/05/2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+@objc(ManagedCache)
+class ManagedCache: NSManagedObject {
+	@NSManaged var timestamp: Date
+	@NSManaged var feed: NSOrderedSet
+
+	internal static func newSingleCache(in context: NSManagedObjectContext) throws -> ManagedCache {
+		try findFirst(in: context).map(context.delete)
+		return ManagedCache(context: context)
+	}
+
+	static func findFirst(in context: NSManagedObjectContext) throws -> ManagedCache? {
+		let request = NSFetchRequest<ManagedCache>(entityName: ManagedCache.entity().name!)
+		return try context.fetch(request).first
+	}
+
+	var localFeed: [LocalFeedImage] {
+		feed.map { $0 as! ManagedFeedImage }
+			.map { LocalFeedImage(id: $0.id, description: $0.imageDescription, location: $0.location, url: $0.url) }
+	}
+
+	static func from(feed: [LocalFeedImage], timestamp: Date, in context: NSManagedObjectContext) -> ManagedCache {
+		let cache = ManagedCache(context: context)
+		cache.timestamp = timestamp
+		cache.feed = ManagedFeedImage.images(from: feed, in: context)
+		return cache
+	}
+}
+
+@objc(ManagedFeedImage)
+class ManagedFeedImage: NSManagedObject {
```

Comment:
This class is not intended to be inherited thus it should be `final`, please make it `final`

```suggestion
final class ManagedFeedImage: NSManagedObject {
```